flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -306,12 +308,16 @@ public AbstractYarnClusterDescriptor createDescriptor(String defaultApplicationN
 		if (cmd.hasOption(JM_MEMORY.getOpt())) {
 			int jmMemory = Integer.valueOf(cmd.getOptionValue(JM_MEMORY.getOpt()));
 			yarnClusterDescriptor.setJobManagerMemory(jmMemory);
+		} else if (config.containsKey(ConfigConstants.JOB_MANAGER_HEAP_MEMORY_KEY)) {
+			yarnClusterDescriptor.setJobManagerMemory(config.getInteger(ConfigConstants.JOB_MANAGER_HEAP_MEMORY_KEY, ConfigConstants.DEFAULT_JOB_MANAGER_HEAP_MEMORY));
```

Comment:
I'm wondering whether the `yarnClusterDescriptor` should not have been automatically instantiated with this value if it gets the configuration as a constructor parameter.